learn/cookbooks/laravel-multitenancy.mdx

```diff
@@ -0,0 +1,365 @@
+- A search API key UID — retrieve it using the [keys endpoints](/references/api/keys)
+
+## Use case
+
+Let's use the example of a customer relationship manager (CRM) application that allows users to store contacts.
+
+Our CRM is a multitenant application, where each user can only access data belonging to their organization. In our example, this means that users will only be able to search contacts of their organization.
+
+On a technical level, this means:
+- A `User` model that belongs to an `Organization`
+- A `Contact` model that belongs to an `Organization` (can only be accessed by users from the same organization)
+- An `Organization` model that has many `User` and many `Contact`s
+
+Below you will find the Laravel code that defines such relationship between our models.
+
+In `app/Models/Contact.php`:
+
+```php
+<?php
+
+namespace App\Models;
+
+use Laravel\Scout\Searchable;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+class Contact extends Model
+{
+    use Searchable;
+
+    public function organization(): BelongsTo
+    {
+        return $this->belongsTo(Organization::class, 'organization_id');
+    }
+}
+```
+
+In `app/Models/User.php`:
+
+```php
+<?php
+
+namespace App\Models;
+
+use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Notifications\Notifiable;
+use Laravel\Sanctum\HasApiTokens;
+
+class User extends Authenticatable
+{
+    use HasApiTokens, Notifiable;
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array<int, string>
+     */
+    protected $fillable = [
+        'name',
+        'email',
+        'password',
+    ];
+
+    /**
+     * The attributes that should be hidden for serialization.
+     *
+     * @var array<int, string>
+     */
+    protected $hidden = [
+        'password',
+        'remember_token',
+    ];
+
+    /**
+     * The attributes that should be cast.
+     *
+     * @var array<string, string>
+     */
+    protected $casts = [
+        'email_verified_at' => 'datetime',
+        'password' => 'hashed',
+    ];
+
+    public function organization()
+    {
+        return $this->belongsTo(Organization::class, 'organization_id');
+    }
+}
+```
+
+And in `app/Models/Organization.php`:
+
+```php
+<?php
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+class Organization extends Model
+{
+    public function contacts(): HasMany
+    {
+        return $this->hasMany(Contact::class);
+    }
+}
+```
+
+Now that we've taken a look at our models and their relationships, let's see how to generate tenant tokens.
+
+## Generating tenant tokens
+
+We want to generate a tenant token for each organization. This token will be used to authenticate requests to Meilisearch and ensure that users can only access data from their organization. All `User` within the same `Organization` will share the same token.
```

Comment:
Currently, all `User`s can search through data belonging to all `Organization`s. To prevent that from happening, you need to generate a tenant token for each organization. You can then use this token to authenticate requests to Meilisearch and ensure that users can only access data from their organization. All `User` within the same `Organization` will share the same token.